Problem: chain-main can't be built for windows

```
 failed after 216.97s error=failed to build for windows_amd64: # github.com/crypto-org-chain/chain-main/v1/cmd/chain-maind/app
111
cmd/chain-maind/app/app.go:160:10: undefined: renameio.WriteFile
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -16,7 +16,7 @@ jobs:
     
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest, windows-latest]
+        os: [ubuntu-latest, windows-latest]
 
     timeout-minutes: 5
     if: >-
@@ -80,11 +80,6 @@ jobs:
       - name: build
         if: | 
           startsWith(matrix.os, 'ubuntu') &&
-          (env.GIT_DIFF != '' || github.event_name == 'issue_comment')
-        run: make build
-      - name: buildmac
-        if: |
-          startsWith(matrix.os, 'mac') &&
           (env.GIT_DIFF != '' || github.event_name == 'issue_comment')
         run: make build
       - name: buildwindows

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -12,7 +12,12 @@ on:
 jobs:
   build:
     name: Build
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+
     timeout-minutes: 5
     if: >-
       github.event_name == 'push' || github.event_name == 'pull_request' || 
@@ -73,8 +78,20 @@ jobs:
         run: |
           echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
       - name: build
+        if: | 
+          startsWith(matrix.os, 'ubuntu') &&
+          (env.GIT_DIFF != '' || github.event_name == 'issue_comment')
         run: make build
-        if: "env.GIT_DIFF != '' || github.event_name == 'issue_comment'"
+      - name: buildmac
+        if: |
+          startsWith(matrix.os, 'mac') &&
+          (env.GIT_DIFF != '' || github.event_name == 'issue_comment')
+        run: make build
+      - name: buildwindows
+        if: |
+          startsWith(matrix.os, 'windows') &&
+          (env.GIT_DIFF != '' || github.event_name == 'issue_comment')
+        run: make buildwindows  
       - name: test
         run: make test
         if: "env.GIT_DIFF != '' || github.event_name == 'issue_comment'"

--- a/Makefile
+++ b/Makefile
@@ -45,6 +45,11 @@ install: check-network go.sum
 
 build: check-network go.sum
 		go build -mod=readonly $(BUILD_FLAGS) $(BUILD_TAGS) -o $(BUILDDIR)/chain-maind ./cmd/chain-maind
+
+buildwindows: check-network go.sum
+		go build -buildmode=exe -mod=readonly $(BUILD_FLAGS) $(BUILD_TAGS) -o $(BUILDDIR)/chain-maind ./cmd/chain-maind
+
+
 .PHONY: build
 
 build-linux: go.sum

--- a/cmd/chain-maind/app/app.go
+++ b/cmd/chain-maind/app/app.go
@@ -38,7 +38,6 @@ import (
 	"github.com/crypto-org-chain/chain-main/v1/app/params"
 	"github.com/crypto-org-chain/chain-main/v1/config"
 	chainmaincli "github.com/crypto-org-chain/chain-main/v1/x/chainmain/client/cli"
-	"github.com/google/renameio"
 )
 
 // NewRootCmd creates a new root command for chain-maind. It is called once in the
@@ -157,7 +156,7 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 			return err
 		}
 
-		return renameio.WriteFile(cleanedPath, bz, 0600)
+		return WriteFile(cleanedPath, bz, 0600)
 	}
 
 	rootCmd.AddCommand(

--- a/cmd/chain-maind/app/appunix.go
+++ b/cmd/chain-maind/app/appunix.go
@@ -1,0 +1,13 @@
+// +build linux darwin
+
+package app
+
+import (
+	"os"
+
+	"github.com/google/renameio"
+)
+
+func WriteFile(filename string, data []byte, perm os.FileMode) error {
+	return renameio.WriteFile(filename, data, perm)
+}

--- a/cmd/chain-maind/app/appwin.go
+++ b/cmd/chain-maind/app/appwin.go
@@ -1,0 +1,12 @@
+// +build windows
+
+package app
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+func WriteFile(filename string, data []byte, perm os.FileMode) error {
+	return ioutil.WriteFile(filename, data, perm)
+}

